### PR TITLE
Add 0004-Fix-concurrent-vector-on-win.patch

### PR DESCRIPTION
**Pull request by Tobias-Fischer**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by Tobias-Fischer** [comment]
This patch works around the following error:
```
%PREFIX%\library\include\tbb\concurrent_vector.h(680): error C2061: syntax error: identifier 'concurrent_vector<`template-type-parameter-1',`template-type-parameter-2'>'
%PREFIX%\library\include\tbb\concurrent_vector.h(1167): note: see reference to class template instantiation 'tbb::concurrent_vector<T,A>' being compiled
C:\Miniconda\conda-bld\ros_1615857806730\_h_env\Library\include\boost/bind/placeholders.hpp(54): note: see reference to class template instantiation 'boost::arg<9>' being compiled
C:\Miniconda\conda-bld\ros_1615857806730\_h_env\Library\include\boost/bind/placeholders.hpp(53): note: see reference to class template instantiation 'boost::arg<8>' being compiled
C:\Miniconda\conda-bld\ros_1615857806730\_h_env\Library\include\boost/bind/placeholders.hpp(52): note: see reference to class template instantiation 'boost::arg<7>' being compiled
C:\Miniconda\conda-bld\ros_1615857806730\_h_env\Library\include\boost/bind/placeholders.hpp(51): note: see reference to class template instantiation 'boost::arg<6>' being compiled
C:\Miniconda\conda-bld\ros_1615857806730\_h_env\Library\include\boost/bind/placeholders.hpp(50): note: see reference to class template instantiation 'boost::arg<5>' being compiled
C:\Miniconda\conda-bld\ros_1615857806730\_h_env\Library\include\boost/bind/placeholders.hpp(49): note: see reference to class template instantiation 'boost::arg<4>' being compiled
C:\Miniconda\conda-bld\ros_1615857806730\_h_env\Library\include\boost/bind/placeholders.hpp(48): note: see reference to class template instantiation 'boost::arg<3>' being compiled
C:\Miniconda\conda-bld\ros_1615857806730\_h_env\Library\include\boost/bind/placeholders.hpp(47): note: see reference to class template instantiation 'boost::arg<2>' being compiled
C:\Miniconda\conda-bld\ros_1615857806730\_h_env\Library\include\boost/bind/placeholders.hpp(46): note: see reference to class template instantiation 'boost::arg<1>' being compiled
%PREFIX%\library\include\tbb\concurrent_vector.h(681): error C2334: unexpected token(s) preceding ':'; skipping apparent function body
%PREFIX%\library\include\tbb\concurrent_vector.h(61): fatal error C1075: '{': no matching token found
```
The upstream bug is described in https://github.com/oneapi-src/oneTBB/issues/274, and a fix for TBB2020 was proposed in https://github.com/oneapi-src/oneTBB/pull/299. I modified the patch so it works with TBB2021 in https://github.com/oneapi-src/oneTBB/pull/362

File: recipe/meta.yaml [hunk 1]
```diff
@@ -34,6 +34,7 @@ source:
     - 0001-CMake-fix-Clang-compiler-detection-on-macOS.patch  # [osx]
     - 0002-CMake-fix-tbb4py-installation.patch
     - 0003-Fix-cross-build-for-arm64-on-macOS.patch           # [osx]
+    - 0004-Fix-concurrent-vector-on-win.patch                 # [win]
 
 build:
   number: 0
```
Comment (AlexVeprev) [inline recipe/meta.yaml:40]: Looks like you forgot to bump build number

**Comment by AlexVeprev** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml (hunk 1 above)
Comment (Tobias-Fischer) [reply recipe/meta.yaml]: Thanks for catching that - done.

**Comment by AlexVeprev** [comment]
@conda-forge-admin, please rerender

**Comment by AlexVeprev** [comment]
Thank you, @Tobias-Fischer!

**Pull request merged**